### DISCUSSION TITLE
Cannot fetch Oracle Sequence via trino

### DISCUSSION THREAD

**Discussion by danielsunzhongyuan** [trinodb/trino#13873] [Q&A] [answered]
In Oracle, we use sequence to generate an incremental id like below:

> create sequence test_seq minvalue 1 maxvalue 9999999999 increment by 1 start with 1;
> create table test_table (
id number default test_seq.nextval not null);

Then I found that this id could not be reached by Trino. With the help of DBVisualizer, we can obviously find that the id does not show up in the columns of test_table.
Besides, below error occurs:
> DBVis error
> An internal error occurred in:
> java.lang.IllegalStateException:Row data at 0/0 is null!
> The error may have affected the application state. Please  to vendor.
> Long Message:
> Row data at 0/0 is null!

trino-jdbc-357.jar
Oracle 19c Enterprise Edition Release 19.0.0.0.0
DbVisualizer Free 13.0.3 (build: 3265)

Could anyone help on this? Thanks.

**Comment by danielsunzhongyuan** [comment]
![Uploading IMG_20220825_113713.jpg…]()

**Comment by ebyhr** [comment, marked as answer]
The cause is `number` type rather than `sequence`. 

I would recommend reading https://trino.io/docs/current/connector/oracle.html#number-to-decimal-configuration-properties. Unsupported type wouldn't appear in the connector. Also, I would recommend upgrading the Trino version.

**Comment by danielsunzhongyuan** [comment]
Thanks for your reply. 
fixed it.
